# Metatranscriptomics
This script takes metatranscriptomic reads, maps them to a fasta file and calculates the FPKM value using cufflinks
-for allowing for Multi-Mapped reads https://sites.google.com/site/princetonhtseq/tutorials/rna-seqWhen mapping reads to the genome or transcriptome, many reads map best to one particular location.  However, some map equally well to one or more places, usually due to repetitive regions in the genome (multiple copies of genes, etc.).  Most mapping algorithm allow to control what happens with these "multi-mappers". By default, TopHat will map a read in up to 20 places in the genome. Reads that map to more locations are discarded. When calculating coverage, the count is divided by the number of places a read mapped to.  For example, a read that maps to two places will contribute 0.5 read to the count for each position. template for Deseq: https://gist.github.com/stephenturner/f60c1934405c127f09a6-run /home2/projects/Wetlands/Metatranscripts/Population_Level_Update through deseq per email Jordan -run /home2/projects/Wetlands/Metatranscripts/Population_Level_3orLess through deseq per email Jordan cd /home2/projects/Wetlands/Metatranscripts/Population_Level_3orLess -make rounded FPKM table time python /ORG-Data/scripts/Nov_Mud_map_trans_to_assembly.py --make_rounded_table  rounded_corrected_MethanosaetaCombo_FPKM_table.txt --folder _cufflinks_corrected_MethanosaetaCombo_ForDatabase.fasta -make raw reads table time python /ORG-Data/scripts/Nov_Mud_map_trans_to_assembly.py --make_raw_reads_table  corrected_MethanosaetaCombo_raw_reads_table.txt (5:05 - 73 min )(8 min for 1 sample)  running in screen -r 19999.pts-4.chimera header lines =  7489 data lines =  0 mapped reads =  32070 Jordan will run files through Deseg2
